mcs51 peephole: add liveness guards to fix unsafe existing rules

Several existing mcs51 rules replace a sequence with a shorter one that
does not preserve the accumulator or carry, without first checking that
they are dead.  Add the missing notUsed()/notVolatile() guards.

	rule 131   - subb a,#0x01 replaced by dec without carry guard
	rule 265   - cpl c / mov bit,c loses carry
	rule 167   - cpl bit removes mov c,bit side effect
	rule 143.a - rlc->rl loses carry
	rule 143.b - rrc->rr loses carry
	rule 174.a - subb-based dec clobbers A and C
	rule 174.b - inc clobbers neither A nor C but original does
	rule 113.a - cjne clobbers C, setb clobbers A
	rule 253.b - djnz clobbers A (was mov a,%1 / dec a)
	rule 264   - cjne clobbers A
	rule 271   - ternary rewrite clobbers A

--- mcs51/peeph.def	2026-07-02 19:21:44.795923900 +0200
+++ mcs51/peeph.def	2026-07-02 19:22:00.972764900 +0200
@@ -402,7 +402,7 @@
 	cjne	%1,%2,%3
 	setb	%4
 %3:
-} if labelRefCount %3 1
+} if labelRefCount(%3 1), notUsed('a' 'c')
 
 replace {
 	clr	a
@@ -788,7 +788,7 @@
 	;	Peephole 131	optimized decrement (not caring for c)
 	dec	%1
 	mov	%2,%1
-}
+} if notUsed('a' 'c')
 
 // ideally the optimizations of rules 132.x should be done in genCmpXX
 replace {
@@ -1004,7 +1004,7 @@
 } by {
 	;	Peephole 143.a	converted rlc to rl
 	rl	a
-}
+} if notUsed('c')
 
 replace {
 	rrc	a
@@ -1012,7 +1012,7 @@
 } by {
 	;	Peephole 143.b	converted rrc to rr
 	rr	a
-}
+} if notUsed('c')
 
 replace {
 	clr	c
@@ -1264,7 +1264,7 @@
 } by {
 	;	Peephole 167	removed redundant bit moves (c not set to %1)
 	cpl	%1
-}
+} if notUsed('c')
 
 replace {
 	jnb	%1,%2
@@ -1356,7 +1356,7 @@
 	mov	r%1,%2
 	;	Peephole 174.a	optimized decrement (acc not set to %2, flags undefined)
 	dec	%2
-}
+} if notUsed('a' 'c')
 
 replace {
 	mov	r%1,%2
@@ -1367,7 +1367,7 @@
 	mov	r%1,%2
 	;	Peephole 174.b	optimized increment (acc not set to %2, flags undefined)
 	inc	%2
-}
+} if notUsed('a' 'c')
 
 replace {
 	mov	%1,@r%2
@@ -4006,7 +4006,7 @@
 } by {
 	;	Peephole 253.b	optimized decrement with compare
 	djnz	%1,%2
-} if notVolatile(%1), operandsNotRelated(%1 '@r0' '@r1')
+} if notVolatile(%1), operandsNotRelated(%1 '@r0' '@r1'), notUsed('a')
 
 
 // applies to f.e. funptrs.c
@@ -4912,7 +4912,7 @@
 	;	Peephole 264	jump optimization (acc not set)
 	cjne	%1,%2,%4
 %3:
-} if labelRefCount(%3 1), labelRefCountChange(%3 -1)
+} if labelRefCount(%3 1), labelRefCountChange(%3 -1), notUsed('a')
 
 
 replace {
@@ -4922,7 +4922,7 @@
 	;	Peephole 265	optimized mov/cpl sequence (carry differs)
 	cpl	c
 	mov	%1,c
-} if notVolatile(%1)
+} if notVolatile(%1), notUsed('c')
 
 replace {
 	mov	%1,c
@@ -5009,7 +5009,7 @@
 %1:
 	mov	%2,a
 %3:
-} if operandsNotRelated('a' 'dptr' %2), labelRefCount(%1 1), labelRefCountChange(%3 -1)
+} if operandsNotRelated('a' 'dptr' %2), labelRefCount(%1 1), labelRefCountChange(%3 -1), notUsed('a')
 
 
 // left-shifting 8 bit to 16 bit result
